File: Serial OrderWise/LeetCode 78 - Subsets/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans ;
        vector<int> temp;
        generate(ans, temp, 0, nums);
        return ans;
    }
    void generate(vector<vector<int>>& ans, vector<int>&temp, int idx, vector<int>&nums) {
        if(idx == nums.size()) {
            ans.push_back(temp);
            return ;
        }
        temp.push_back(nums[idx]);
        generate(ans, temp, idx+1, nums);
        temp.pop_back();
        generate(ans, temp, idx+1, nums);
    }
};

File: Serial OrderWise/LeetCode 78 - Subsets/Solution.py
class Solution(object):
    def generate(self, ans, temp, idx, nums) :
        if idx == len(nums) :
            ans.append(list(temp))
            return
        temp.append(nums[idx])
        self.generate(ans, temp, idx+1, nums)
        temp.pop()
        self.generate(ans, temp, idx+1, nums)

    def subsets(self, nums):
        ans, temp = [],[]
        self.generate(ans,temp,0,nums)
        return ans